packages/plugin-layout/src/ZoneNode.ts

```diff
@@ -0,0 +1,36 @@
+import { VNode } from '../../core/src/VNodes/VNode';
+
+export type ZoneIdentifier = string;
+
+export class ZoneNode extends VNode {
+    hidden: Map<VNode, boolean> = new Map();
+
+    constructor(public managedZones: ZoneIdentifier[]) {
+        super();
+    }
+
+    get name(): string {
+        return super.name + ': ' + this.managedZones.join();
+    }
+
+    hide(child: VNode): void {
+        this.hidden.set(child, true);
+        return;
+    }
+    show(child: VNode): void {
+        this.hidden.set(child, false);
+        const parentZone: ZoneNode = this.ancestor(ZoneNode);
+        if (parentZone) {
+            parentZone.show(this);
+        }
+    }
+    _insertAtIndex(child: VNode, index: number): void {
+        this.hidden.set(child, true);
```

Comment:
Strange behavior, why whould you change the visibility of a node when you move it?
It's like having div.display = 'block' when moving the div no?

Reply:
This is the area that knows what to do with the elements to hide (put them in an onlget (eg: a tabbed panel), not add them to the dom (ex: dialog), show them anyway so as not to have a empty area (ex: hand)